### A. Inventaire des Sources de Données

In [17]:
import pandas as pd
import os

Chargement de données 

In [20]:
RAW_DATA_DIR = os.path.join("..","data", "raw")

In [19]:
# Définition des métadonnées contextuelles
metadata_sources = {
    'authentication_logs.csv': {
        'propriétaire': 'Équipe Infrastructure / Réseau (AD, VPN, RDP)',
        'contenu': "Événements d'authentification (tentatives, réussites/échecs, IP, machines)",
        'fréquence': 'Flux continu / Temps réel'
    },
    'edr_alerts.csv': {
        'propriétaire': 'Équipe SOC / Sécurité Opérationnelle',
        'contenu': "Alertes de sécurité EDR (processus, sévérité, statut, décision analyste)",
        'fréquence': 'Temps réel sur détection'
    },
    'assets.csv': {
        'propriétaire': 'Équipe IT / Gestion de parc',
        'contenu': "Référentiel des actifs physiques et virtuels (postes, serveurs, VM)",
        'fréquence': 'Statique / Mise à jour ponctuelle'
    },
    'users.csv': {
        'propriétaire': 'Équipe RH / Équipe IT(IAM)',
        'contenu': "Référentiel des utilisateurs (département, rôle, statut, privilèges)",
        'fréquence': 'Statique / Mise à jour périodique'
    }
}

print("=== RAPPORT D'INVENTAIRE TECHNIQUE DES SOURCES ===")

inventaire_data = []

for fichier, info in metadata_sources.items():
    file_path = os.path.join(RAW_DATA_DIR, fichier)
    if os.path.exists(file_path):
        df = pd.read_csv(file_path)
        inventaire_data.append({
            'Fichier': fichier,
            'Propriétaire': info['propriétaire'],
            'Contenu': info['contenu'],
            'Format': 'CSV (Tabulaire)',
            'Volume': f"{df.shape[0]} lignes, {df.shape[1]} cols",
            'Fréquence': info['fréquence'],
            'Qualité Initiale': f"Doublons: {df.duplicated().sum()} | Nulls: {df.isnull().sum().sum()}"
        })
    else:
        print(f"⚠️ Fichier introuvable : {fichier}")

# Affichage structuré sous forme de DataFrame pour le notebook
df_inventaire = pd.DataFrame(inventaire_data)
display(df_inventaire)

=== RAPPORT D'INVENTAIRE TECHNIQUE DES SOURCES ===


,Fichier,Propriétaire,Contenu,Format,Volume,Fréquence,Qualité Initiale
0,authentication_logs.csv,"Équipe Infrastructure / Réseau (AD, VPN, RDP)","Événements d'authentification (tentatives, réu...",CSV (Tabulaire),"4992 lignes, 9 cols",Flux continu / Temps réel,Doublons: 145 | Nulls: 151
1,edr_alerts.csv,Équipe SOC / Sécurité Opérationnelle,"Alertes de sécurité EDR (processus, sévérité, ...",CSV (Tabulaire),"1929 lignes, 9 cols",Temps réel sur détection,Doublons: 74 | Nulls: 234
2,assets.csv,Équipe IT / Gestion de parc,Référentiel des actifs physiques et virtuels (...,CSV (Tabulaire),"200 lignes, 7 cols",Statique / Mise à jour ponctuelle,Doublons: 0 | Nulls: 34
3,users.csv,Équipe RH / Équipe IT(IAM),"Référentiel des utilisateurs (département, rôl...",CSV (Tabulaire),"400 lignes, 6 cols",Statique / Mise à jour périodique,Doublons: 5 | Nulls: 0


L'inventaire technique des quatre sources de données brutes:

* authentication_logs.csv
    - Qualité initiale : Moyenne ; présence de 145 doublons stricts, de 151 valeurs manquantes et d'extractions hétérogènes (timestamp variables et adresses IP parfois mal formées).

* edr_alerts.csv
    - Qualité initiale : Faible à modérée ; présence de 74 doublons stricts et de 234 valeurs manquantes (principalement sur le champ analyst_decision, qui n'est renseigné que si l'alerte a été traitée manuellement)

* assets.csv
    - Qualité initiale : Incertaine ; 0 doublon strict, mais présence de 34 valeurs manquantes et absence de garantie de mise à jour par l'IT (risque d'obsolescence).

* users.csv
    - Qualité initiale : Incertaine ; 5 doublons détectés, aucune valeur manquante brute, mais référentiel non garanti à jour (risque de décalage avec les connexions réelles).
